server/zoom/www/handlers/tools_refactor_paths.py

```diff
@@ -18,51 +18,81 @@ def zk(self):
     def configuration(self):
         return self.application._configuration
 
+    @property
+    def app_state_path(self):
+        return self.configuration._application_state_path + '/'
+
+    @property
+    def agent_config_path(self):
+        return self.configuration._agent_configuration_path[:-1]
+
+    @property
+    def old_path(self):
+        logging.info('Whats the old path')
+        return self.get_argument('oldPath')
+
+    @property
+    def new_path(self):
+        return self.get_argument('newPath')
 
     @TimeThis(__file__)
     def post(self):
         """
         Save filter
         """
-        config_dict = dict()
-        comp_id_found = False
-        app_state_path = self.configuration._application_state_path + '/'
-        agent_config_path = self.configuration._agent_configuration_path[:-1]
-
-
-        old_path = self.get_argument('oldPath')
-        new_path = self.get_argument('newPath')
-
-
-        old_uid = old_path.replace(app_state_path, '')
-        new_uid = new_path.replace(app_state_path, '')
-
-        comp_id_string = 'id="{0}"'.format(old_uid)
-
         # check if the path exists
-        if not self.zk.exists(old_path):
+        if not self.zk.exists(self.old_path):
             self.set_status(httplib.NOT_FOUND)
             self.write({'errorText': 'Old path does not exist'})
             return
+        #if the old path has children or not
+        children = self.zk.get_children(self.old_path)
+        if not children:
+            self._test_partial_path_refactor()
+        else:
+            # Take the first child and check if its ephemeral or not
```

Comment:
We don't have state instances in Zookeeper that have combinations of ephemeral and static nodes under a path